### PR TITLE
drivers: sensor: lsm303dlhc magn: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

File: samples/sensor/lsm303dlhc/src/main.c [hunk 1]
```diff
@@ -38,20 +38,16 @@ static int32_t read_sensor(const struct device *sensor,
...
-	const struct device *magnetometer = device_get_binding(
-						DT_LABEL(DT_INST(0, st_lsm303dlhc_magn)));
+	const struct device *accelerometer = DEVICE_DT_GET_ONE(st_lis2dh);
+	const struct device *magnetometer = DEVICE_DT_GET_ONE(st_lsm303dlhc_magn);
 
-	if (accelerometer == NULL) {
-		printf("Could not get %s device\n",
-				DT_LABEL(DT_INST(0, st_lis2dh)));
+	if (!device_is_ready(accelerometer)) {
+		printf("Could not get accelerometer device\n");
```
Comment (MaureenHelm) [inline samples/sensor/lsm303dlhc/src/main.c:45]: Update the message

File: samples/sensor/lsm303dlhc/src/main.c [hunk 2]
```diff
@@ -38,20 +38,16 @@ static int32_t read_sensor(const struct device *sensor,
...
 		return;
 	}
 
-	if (magnetometer == NULL) {
-		printf("Could not get %s device\n",
-				DT_LABEL(DT_INST(0, st_lsm303dlhc_magn)));
+	if (!device_is_ready(magnetometer)) {
+		printf("Could not get magnetometer device\n");
```
Comment (MaureenHelm) [inline samples/sensor/lsm303dlhc/src/main.c:50]: ditto

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/sensor/lsm303dlhc/src/main.c (hunk 1 above)
Comment (benjaminbjornsson) [reply samples/sensor/lsm303dlhc/src/main.c]: Done!

File: samples/sensor/lsm303dlhc/src/main.c (hunk 2 above)
Comment (benjaminbjornsson) [reply samples/sensor/lsm303dlhc/src/main.c]: Done!

**Review by galak — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**